### ISSUE TITLE
Oracle: dialect does not support power and mod operators

### ISSUE THREAD

**Issue by saulotoledo** [sqlfluff/sqlfluff#7060]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff fails to parse Oracle-specific arithmetic operators `**` (power) and `MOD`, causing parse errors when these operators are used in SQL code. These operators are standard in Oracle SQL but not part of the ANSI specification.

### Expected Behaviour

Oracle-specific arithmetic operators should parse correctly in all expression contexts:

```sql
BEGIN
  -- Power operator in IF statements
  IF i ** 2 > 50 THEN
    DBMS_OUTPUT.PUT_LINE('i squared is greater than 50');
  END IF;

  -- MOD operator in IF statements
  IF i MOD 2 = 0 THEN
    DBMS_OUTPUT.PUT_LINE('i is even number');
  END IF;

  -- Power operator in SELECT expressions
  SELECT
    id,
    value ** 2 AS value_squared,
    amount ** 0.5 AS square_root
  FROM test_table;

  -- MOD operator in SELECT expressions
  SELECT
    id,
    amount MOD 100 AS last_two_digits,
    CASE WHEN id MOD 2 = 0 THEN 'Even' ELSE 'Odd' END AS parity
  FROM test_table;

  -- Combined usage in complex expressions
  SELECT ((a + b) ** 2) MOD 1000 AS result FROM dual;
END;
```

### Observed Behaviour

SQLFluff raises parse errors when encountering these operators and fails processing the `.sql` files:

```sql
-- Parse error: Expected one of: 'EOF' or 'create', 'select', etc.
IF i ** 2 > 50 THEN
     ^^
-- Parse error: Expected one of: 'EOF' or 'create', 'select', etc.  
IF i MOD 2 = 0 THEN
     ^^^
```

### How to reproduce

1. Create an Oracle SQL file (e.g., oracle_operators.sql) with the following content:

```sql
BEGIN
  -- Power operator test
  IF i ** 2 > 50 THEN
    DBMS_OUTPUT.PUT_LINE('Power test');
  END IF;
  
  -- MOD operator test
  IF i MOD 2 = 0 THEN
    DBMS_OUTPUT.PUT_LINE('MOD test');
  END IF;
  
  -- In SELECT expressions
  SELECT value ** 2, id MOD 10 FROM test_table;
END;
```

2. Configure sqlfluff for Oracle dialect (see config section).
3. Run sqlfluff to parse/lint the file:

```sh
sqlfluff lint oracle_operators.sql --dialect oracle
```

4. Observe parse errors for both `**` and `MOD` operators.

### Dialect

oracle

### Version

3.4.2

### Configuration

```ini
[sqlfluff]
dialect = oracle

[sqlfluff:core]
max_line_length = 120

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 2
```

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
